dev-python/spyder-5.0 version bump

AndrewAmmerlaan opened the pull request:
Version bumps spyder to 5.0 alpha version, adds some new pytest/sphinx deps, and adds py3.8 support (thanks to @mattst88  for patching py3.8 support into python-language-server in https://github.com/gentoo/gentoo/pull/16732)

Also found a bug in dev-python/binaryornot where the version number is detected wrong, I fixed it, but my revision bump should probably be bumped to stable immediately (which I cannot do without the QA bot complaining), so it would be great if a dev could do that for me.

sbraz commented:
That is *a lot* of new packages. Also, why argparse? It's in the stdlib since 2.7.

AndrewAmmerlaan commented:
> That is _a lot_ of new packages. 

Sorry about that, a major version bump of a high level python packages (like spyder) tends to add lots of new packages. In this case spyder-5 now depends on cookiecutter (which had to be keyworded) which depends on binaryornot (which had a bug that needed fixing), and the spyder-5 tests now depend on pytest-ordering and pytest-xvfb which in turn adds most of the new packages here, the spyder docs also changed and now depend on sphinx-panels and the pydata-sphinx-theme both of which are new packages.

> Also, why argparse? It's in the stdlib since 2.7.

entrypoint2 specified it as a dependency, I had not realized that this was already in the stdlib. The tests on entrypoint2 passed after I removed argparse, so I have dropped dev-python/argparse from the PR.

juippis commented on dev-python/spyder/spyder-4.1.5.ebuild:
```diff
@@ -0,0 +1,201 @@
+	>=dev-python/pygments-2.0[${PYTHON_USEDEP}]
+	>=dev-python/pylint-1.0[${PYTHON_USEDEP}]
+	>=dev-python/python-language-server-0.34.0[${PYTHON_USEDEP}]
+		dev-python/autopep8[${PYTHON_USEDEP}]
+		>=dev-python/flake8-3.8.0[${PYTHON_USEDEP}]
+		>=dev-python/mccabe-0.6.0[${PYTHON_USEDEP}]
+		<dev-python/mccabe-0.7.0[${PYTHON_USEDEP}]
+		>=dev-python/pycodestyle-2.6.0[${PYTHON_USEDEP}]
+		<dev-python/pycodestyle-2.7.0[${PYTHON_USEDEP}]
+		>=dev-python/pydocstyle-2.0.0[${PYTHON_USEDEP}]
+		>=dev-python/pyflakes-2.2.0[${PYTHON_USEDEP}]
+		<dev-python/pyflakes-2.3.0[${PYTHON_USEDEP}]
+		dev-python/pylint[${PYTHON_USEDEP}]
+		>=dev-python/rope-0.10.5[${PYTHON_USEDEP}]
+		dev-python/yapf[${PYTHON_USEDEP}]
```
What's with the weird indentation? Also please sort somehow.

AndrewAmmerlaan replied:
This is to make things more clear for myself. Spyder depends on python-language-server[all] where the [all] part would pull in the list above on pip. Because we do not have USE flags for optional runtime dependencies on python-language-server, my solution was to translate the [all] part to the above list and indent it a bit more to clearly differentiate them from the other dependencies. I'll add a comment in the ebuild to make clear what I did and why.

juippis reviewed (commented):
Overall the only blocker in here in my opinion is the _alpha version, and follow-up commits regarding that. Other comments, from what I remember at least, are mostly nitpicking.

For many lists where I've asked to "sort" you can usually sort these within your editor, most editors are capable of that if there are no `< > = ~` in the beginning.

Overall it looks good.

AndrewAmmerlaan commented on dev-python/vncdotool/Manifest:
```diff
@@ -0,0 +1,2 @@
+DIST vncdotool-0.13.0.tar.gz 72780 BLAKE2B 6a27565c5410148f569bd5a8e212327d734f9fd044549683bafcbbaed0df56e0a4853b4750eadc74e2186463313cd510c5290a8c4de69eba5ecbc42ea5d94c32 SHA512 65e275c53dbc805114801b149bff86743cdcb2b8d69b038425fd7dd6f911b113e332246178080c71314d58624137152205057c00b451a3ec6b46541f1b294b98
+DIST vncdotool-1.0.0.tar.gz 62580 BLAKE2B d09b26e81496055f17e96fa5b60ea006ddd210fa8b0044785c107582af18a61fb5beec83f577222b32cc8f446694235cd3ab468cfa917ec3ef4b00b6fbc1a8bc SHA512 45fba9ba3549ad575923bcfb3ab62c53b3c2d85882c87ca2b656795314b9c32aa989a1fdae280f904cb4d79f509622810ffd4d1e2977beac917f0289d6ff24a4
```
pyvirtualdsplay requires vncdotool-0.13.0, the other version is the latest version.

I figured I'd add the latest version as well, for users who use only vncdotool and not pyvirtualdisplay

AndrewAmmerlaan commented on dev-python/vncdotool/vncdotool-0.13.0.ebuild:
```diff
@@ -0,0 +1,41 @@
+KEYWORDS="~amd64 ~x86"
+
+# No clue why this happens:
+# pexpect.exceptions.ExceptionPexpect: The command was not found or was not executable: vncev.
```
This has been confusing me for days, I cannot figure out what is supposed to provide this vncev. Google seems to suggest libvncserver but installing that didn't work. Maybe we need libvncclient instead? Until I figure out how to fix this, I figured I'd just RESTRICT the tests.

juippis replied:
Yeah we'd have to check the build system and make a patch for ::gentoo package. That's out of scope for this PR, but something you could investigate further.

AndrewAmmerlaan commented on dev-python/sphinx-panels/sphinx-panels-0.4.1.ebuild:
```diff
@@ -0,0 +1,24 @@
+
+DEPEND="
+	dev-python/docutils[${PYTHON_USEDEP}]
+	dev-python/sphinx[${PYTHON_USEDEP}]
```
`distutils_enable_sphinx` adds `BDEPEND="doc? ( dev-python/sphinx )` and this is a theme thing for sphinx so it should also depend on sphinx if USE="-doc".

AndrewAmmerlaan commented on dev-python/spyder-terminal/spyder-terminal-0.4.2.ebuild:
```diff
@@ -0,0 +1,24 @@
+
+DESCRIPTION="Run system terminals inside Spyder"
+HOMEPAGE="https://github.com/spyder-ide/spyder-terminal"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
```
I can't get the github tarball to install through emerge, it requires npm to fetch some things which fails in emerge.

AndrewAmmerlaan commented on dev-python/spyder/Manifest:
```diff
@@ -1,2 +1,4 @@
 DIST spyder-4.1.4.tar.gz 11699453 BLAKE2B 208ba4f32fa548ef8478d72c046e70ba43a2fccd2040629af03f7f56c379ebefa1e90594054f15a92aebdd14307d001c68ee5e6d2401a841e408c4af42436c33 SHA512 8a5f63b11648002838e0b4e5427b1c7cf9cbe4a4fd20af7e33e5e789cddaf6393bb7200bfacf8c7af19d723f12d99d2e962a769706b806335f7146ddd6853735
+DIST spyder-5.0.0_alpha1.tar.gz 11763788 BLAKE2B dde8d5c5de0720ca8cbe1d65d5c3120da7a3e7d84865829eab19065f5cf2f3879bc0a8b11eb47b9799726cf11b8b2dd6a0137ea8aa0aeb1fd20aeae989cdd433 SHA512 25091b499f3cd1fdea225207402620a1cfb81eba0a1501dc3f44dbba3da9eafdfa23658e159058645b894108f7c94bbce59a2c18580e918f465c98e3f411e3ac
```
Well the thing is the 4.1.5 version was released after the 5.0.0_alpha version, which makes me unsure which version to keep and which to drop. Also the patch files in both versions are near identical, it fixes the name of the sphinx theme in spyder-docs. So which do I keep? the version that is more recent, or the one that has the higher version number?

AndrewAmmerlaan commented on dev-python/pytest-faulthandler/pytest-faulthandler-2.0.1.ebuild:
```diff
@@ -0,0 +1,20 @@
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="
+	>=dev-python/pytest-5.0[${PYTHON_USEDEP}]
+"
```
There are no tests here: https://github.com/pytest-dev/pytest-faulthandler

It depends on pytest not for testing itself but because it is a pytest plugin, so I suppose this should be a RDEPEND.

AndrewAmmerlaan commented on dev-python/pydata-sphinx-theme/pydata-sphinx-theme-0.3.1.ebuild:
```diff
@@ -0,0 +1,22 @@
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="dev-python/sphinx[${PYTHON_USEDEP}]"
+
+# ModuleNotFoundError: No module named 'plotly' even if dev-python/plotly is installed
+#distutils_enable_sphinx docs dev-python/recommonmark dev-python/numpydoc dev-python/jupyter-sphinx dev-python/plotly
```
I did not add the test because they introduce yet even more pytest-* packages. And this PR is already huge.

But, alright, I'll add them, I have the ebuilds lying around anyway.

AndrewAmmerlaan commented:
> Overall the only blocker in here in my opinion is the _alpha version, and follow-up commits regarding that. Other comments, from what I remember at least, are mostly nitpicking.
> 
> For many lists where I've asked to "sort" you can usually sort these within your editor, most editors are capable of that if there are no `< > = ~` in the beginning.
> 
> Overall it looks good.

@juippis I dropped the alpha version and made the changes you suggested. I kept the commits that add the spyder<5 dependency to the plugins. Because I expect that when the stable -5.0 version is released that compatibility with the plugins will be broken anyway, and that the plugins will be updated slowly later. Since this is exactly what happened in the -4.0 update.

juippis commented on dev-util/cookiecutter/cookiecutter-1.7.2-r1.ebuild:
```diff
@@ -14,7 +14,7 @@ SRC_URI="https://github.com/audreyr/cookiecutter/archive/${PV}.tar.gz -> ${P}.ta
 LICENSE="BSD"
 SLOT="0"
-KEYWORDS="~amd64"
+KEYWORDS="~amd64 ~x86"
```
For some reason this doesn't apply cleanly on ::gentoo repo anymore, but I'll add the KW after testing it on x86.

juippis replied:
Nevermind, now it did after -3 switch.

AndrewAmmerlaan commented on dev-python/pyscreenshot/pyscreenshot-2.2.ebuild:
```diff
@@ -0,0 +1,72 @@
+	distutils-r1_python_prepare_all
+}
+
+python_test() {
+	virtx pytest -vv
+}
```
That is strange, it works for me without pqiv. Perhaps the test code has some conditional stuff: only do this test if this is installed, I'll look into it.

AndrewAmmerlaan replied:
I had imagemagick installed, very likely it will test whatever media-gfx tool is installed, which explains why I found that the imagemagick test failed, but never found the failing pqiv test.

AndrewAmmerlaan commented on dev-python/pydata-sphinx-theme/pydata-sphinx-theme-0.4.0.ebuild:
```diff
@@ -0,0 +1,25 @@
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+BDEPEND="dev-python/pytest-regressions[${PYTHON_USEDEP}]"
+
```
Thanks :smile:

AndrewAmmerlaan replied:
https://github.com/gentoo/gentoo/pull/17551

AndrewAmmerlaan replied:
Apparently this needs the `distutils_install_for_testing` function: https://github.com/gentoo/gentoo/pull/17608

AndrewAmmerlaan commented on dev-python/mss/mss-6.0.0.ebuild:
```diff
@@ -0,0 +1,31 @@
+distutils_enable_tests pytest
+distutils_enable_sphinx docs/source dev-python/sphinx_rtd_theme
+
+python_test() {
+	virtx pytest -vv
+}
```
Sigh, I really wish upstreams would document their test dependencies better, I'll look into it.

AndrewAmmerlaan replied:
This should fix it: https://github.com/gentoo/gentoo/pull/17550

AndrewAmmerlaan commented on dev-python/easyprocess/easyprocess-0.3.ebuild:
```diff
@@ -0,0 +1,28 @@
+BDEPEND="test? (
+	dev-python/pytest-timeout[${PYTHON_USEDEP}]
+	dev-python/six[${PYTHON_USEDEP}]
+	dev-python/pyvirtualdisplay[${PYTHON_USEDEP}]
```
https://github.com/gentoo/gentoo/pull/17549
